daemon/pruner/pruner.go

```diff
@@ -82,6 +83,13 @@ type PrunerFactory struct {
 	promPruneSecs *prometheus.HistogramVec
 }
 
+type SinglePrunerFactory struct {
```

Comment:
I see why you created `SinglePrunerFactory`: the current `PrunerFactory` doesn't cover single-side pruning, since we wanted to defer that post 0.1 (this functionality is indeed also needed for sink-side pruning).

I'll think about whether the duplication is ok for now.
Until then, keep this as is.